iree/compiler/Dialect/VM/Conversion/VMToEmitC/DropDisabledFunctions.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/VM/Conversion/VMToEmitC/DropDisabledFunctions.h"
+
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace VM {
+
+class DropDisabledFunctionsPass
+    : public PassWrapper<DropDisabledFunctionsPass,
+                         OperationPass<IREE::VM::ModuleOp>> {
+ public:
+  void runOnOperation() override {
+    // Remove exported functions annotated with emitc.disable.
+    SmallVector<Operation *, 4> opsToRemove;
+    getOperation()->walk([&](IREE::VM::ExportOp exportOp) {
```

Comment:
multiple exports may reference the same function, and some functions may be both used internally as well as exported and this would break those (you'd end up with calls to functions that didn't exist)

if instead it was on the export had the attribute then you could just drop that and then symboldce should clean up the vm.func if it is not used internally